hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/sink/PrometheusMetricsSink.java

```diff
@@ -53,42 +57,16 @@ public PrometheusMetricsSink() {
 
   @Override
   public void putMetrics(MetricsRecord metricsRecord) {
-    for (AbstractMetric metrics : metricsRecord.metrics()) {
-      if (metrics.type() == MetricType.COUNTER
-          || metrics.type() == MetricType.GAUGE) {
+    for (AbstractMetric metric : metricsRecord.metrics()) {
+      if (metric.type() == MetricType.COUNTER
+          || metric.type() == MetricType.GAUGE) {
 
         String key = prometheusName(
-            metricsRecord.name(), metrics.name());
-
-        StringBuilder builder = new StringBuilder();
-        builder.append("# TYPE ")
-            .append(key)
-            .append(" ")
-            .append(metrics.type().toString().toLowerCase())
-            .append("\n")
-            .append(key)
-            .append("{");
-        String sep = "";
-
-        //add tags
-        for (MetricsTag tag : metricsRecord.tags()) {
-          String tagName = tag.name().toLowerCase();
-
-          //ignore specific tag which includes sub-hierarchy
-          if (!tagName.equals("numopenconnectionsperuser")) {
-            builder.append(sep)
-                .append(tagName)
-                .append("=\"")
-                .append(tag.value())
-                .append("\"");
-            sep = ",";
-          }
```

Comment:
@Kimahriman 
Before committing your patch, I tested it on my local environment and found an issue.

Would you ignore "numopenconnectionsperuser" tag in your patch? The label value can become `numopenconnectionsperuser="{"hdfs":17}"`, and it is not valid in prometheus.

```
% curl -s http://<namenode>:9870/prom | promtool check metrics                                  
error while linting: text format parsing error in line 21: unexpected end of label value "{"
```

Sorry for the back and forth.

Reply:
Ah I didn't realize that was actually unable to be parsed by prometheus, I thought it was just a thing you definitely don't want included in your prometheus series that could be excluded from scraping. I'll add that back in